test/unit/math/prim/fun/lb_transform_test.cpp

```diff
@@ -3,60 +3,198 @@
+  }
 
-  double lp = 0.0;
-  EXPECT_MATRIX_EQ(resv, stan::math::lb_constrain(input, lbv, lp));
-  EXPECT_EQ(input.sum(), lp);
-  lp = 0.0;
-  EXPECT_MATRIX_EQ(res, stan::math::lb_constrain(input, lb, lp));
-  EXPECT_EQ(input.sum(), lp);
-}
+  // matrix, matrix
+  {
+    Eigen::VectorXd result = stan::math::lb_constrain(x, lb);
+    for(size_t i = 0; i < result.size(); ++i) {
+      EXPECT_FLOAT_EQ(result(i), stan::math::lb_constrain(x(i), lb(i)));
+    }
+    EXPECT_MATRIX_EQ(x, stan::math::lb_free(result, lb));
+    EXPECT_THROW(stan::math::lb_constrain(x, lb_bad), std::invalid_argument);
+  }
 
-TEST(prob_transform, lb_j) {
-  double lp = 15.0;
-  EXPECT_FLOAT_EQ(exp(-1.0) + 2.0, stan::math::lb_constrain(-1.0, 2.0, lp));
-  EXPECT_FLOAT_EQ(15.0 - 1.0, lp);
+  // matrix, real
+  {
+    double lp0 = 0.0;
+    double lp1 = 0.0;
+    Eigen::VectorXd result = stan::math::lb_constrain(x, lbd, lp0);
+    for(size_t i = 0; i < result.size(); ++i) {
+      EXPECT_FLOAT_EQ(result(i), stan::math::lb_constrain(x(i), lbd, lp1));
+    }
+    EXPECT_FLOAT_EQ(lp0, lp1);
+  }
 
-  double lp2 = 8.6;
-  EXPECT_FLOAT_EQ(7.9, stan::math::lb_constrain(
-                           7.9, -std::numeric_limits<double>::infinity(), lp2));
-  EXPECT_FLOAT_EQ(8.6, lp2);
-}
-TEST(prob_transform, lb_f) {
-  EXPECT_FLOAT_EQ(log(3.0 - 2.0), stan::math::lb_free(3.0, 2.0));
-  EXPECT_FLOAT_EQ(
-      1.7, stan::math::lb_free(1.7, -std::numeric_limits<double>::infinity()));
+  // matrix, matrix
+  {
+    double lp0 = 0.0;
+    double lp1 = 0.0;
+    Eigen::VectorXd result = stan::math::lb_constrain(x, lb, lp0);
+    for(size_t i = 0; i < result.size(); ++i) {
+      EXPECT_FLOAT_EQ(result(i), stan::math::lb_constrain(x(i), lb(i), lp1));
+    }
+    EXPECT_FLOAT_EQ(lp0, lp1);
+    EXPECT_THROW(stan::math::lb_constrain(x, lb_bad, lp1), std::invalid_argument);
+  }
 }
 
-TEST(prob_transform, lb_f_exception) {
-  double lb = 2.0;
-  EXPECT_THROW(stan::math::lb_free(lb - 0.01, lb), std::domain_error);
+TEST(prob_transform, lb_constrain_std_vector) {
+  Eigen::VectorXd x(2);
+  x << -1.0, 1.1;
+  Eigen::VectorXd lb(2);
+  lb << 2.0, stan::math::NEGATIVE_INFTY;
+  Eigen::VectorXd lb2(2);
+  lb << stan::math::NEGATIVE_INFTY, 1.0;
+  double lbd = 2.0;
+
+  Eigen::VectorXd lb_bad(3);
+
+  std::vector<Eigen::VectorXd> x_vec = { x, 2 * x };
+  std::vector<Eigen::VectorXd> lb_vec = { lb, lb2 };
+
+  std::vector<Eigen::VectorXd> lb_vec_bad1 = { lb, lb2, lb };
+  std::vector<Eigen::VectorXd> lb_vec_bad2 = { lb, lb_bad };
+
+  // matrix[], real
+  {
+    std::vector<Eigen::VectorXd> result = stan::math::lb_constrain(x_vec, lbd);
+    for(size_t i = 0; i < result.size(); ++i) {
+      EXPECT_MATRIX_EQ(result[i], stan::math::lb_constrain(x_vec[i], lbd));
+    }
+    auto free_x = stan::math::lb_free(result, lbd);
+    for(size_t i = 0; i < result.size(); ++i) {
+      EXPECT_MATRIX_EQ(x[i], free_x[i]);
+    }
+  }
+
+  // matrix[], matrix
+  {
```

Comment:
Why are these in brackets like this?

Reply:
I was just marking off tests with arguments in Stan syntax so I could keep them straight in my head. We can delete these comments, of course. I just left them cause it reminds me what the code is doing.